sdk/communication/communication-identity/src/communicationIdentityClient.ts

```diff
@@ -224,4 +237,16 @@ export class CommunicationIdentityClient {
       }
     );
   }
+
+  private parseOptions(
+    options: CreateUserAndTokenOptions | GetTokenOptions
+  ): CommunicationIdentityIssueAccessTokenOptionalParams {
+    const operationOptions: CommunicationIdentityIssueAccessTokenOptionalParams = options;
+    if (options.tokenExpiresInMinutes && typeof options.tokenExpiresInMinutes === "number") {
```

Comment:
what's the intent of the whole method? just a type check? I don't think we need to do this in typescript. I don't think we usually do this kind of checks. I'd probably get rid of the whole method

Reply:
I guess it was for the prev solution with operations options bag, as we introduced GetTokenOptions & CreateUserAndTokenOptions there is no need for the whole method

Reply:
Yeah, Petr is totally right in his [comment](https://github.com/Azure/azure-sdk-for-js/pull/22623#discussion_r972968641) the API of the service layer is expecting the following options:
Create user and token API => CommunicationIdentityCreateOptionalParams
Get token API => CommunicationIdentityIssueAccessTokenOptionalParams
so @maximrytych-ms you need to delete this function and create requested options in each public API

Reply:
I am using another approach now. I don't need those types at all.